packages/main/src/DayPicker.js

```diff
@@ -637,20 +789,39 @@ class DayPicker extends UI5Element {
 
 	get _maxDate() {
 		if (this.maxDate) {
-			const jsDate = new Date(this.getFormat().parse(this.maxDate).getFullYear(), this.getFormat().parse(this.maxDate).getMonth(), this.getFormat().parse(this.maxDate).getDate());
-			const oCalDate = CalendarDate.fromTimestamp(jsDate.getTime(), this._primaryCalendarType);
-			return oCalDate.valueOf();
+			return this._getTimeStampFromString(this.maxDate);
 		}
-		return this.maxDate;
+
+		const maxDate = new CalendarDate(1, 0, 1, this._primaryCalendarType);
```

Comment:
Why don't you set the year and month here?

Reply:
Same workaround as for the minimum date.